### PR TITLE
`kwok` and `kwokctl` tools

### LINKED ISSUE #991: [Tool] Add kwok tool for K8s load testing
KWOK - [KWOK (Kubernetes WithOut Kubelet)](https://github.com/kubernetes-sigs/kwok/#kwok-kubernetes-without-kubelet) is a tool for creating large "virtual clusters" where Pods are never really created and run, but are simulated instead.

https://github.com/kubernetes-sigs/kwok/#kwok-kubernetes-without-kubelet

It's useful for testing operators and controllers with large amounts of nodes or Custom Resources.

## Expected Behaviour

Add as a new download option using:

https://github.com/kubernetes-sigs/kwok/releases

### PR DESCRIPTION
**Pull request by Shikachuu**
resolves #991

## Description

Add `kwok` and `kwokctl` tools to arkade

## Motivation and Context

- [X] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md)) #991 

## How Has This Been Tested?

`./hack/test-tool.sh` and unit test for both tools.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [X] I have updated the list of tools in README.md if (required) with `./arkade get --format markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`

- [X] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by Shikachuu** [comment]
Hmm...seem like [boz/kail](https://github.com/boz/kail) project's release pipeline is failing, because of that there is no binaries in the new release.

Should I pin a version for that tool in this PR or create a separate one?

**Comment by alexellis** [comment]
@Shikachuu please could you raise an issue with them in their repo?

File: pkg/get/get_test.go [hunk 1]
```diff
@@ -6703,3 +6703,101 @@ func Test_DownloaSkupper(t *testing.T) {
...
+			url:     `https://github.com/kubernetes-sigs/kwok/releases/download/v0.4.0/kwok-darwin-arm64`,
+		},
+		{
+			os:      "darwin",
+			version: toolVersion,
+			arch:    arch64bit,
+			url:     `https://github.com/kubernetes-sigs/kwok/releases/download/v0.4.0/kwok-darwin-amd64`,
+		},
+		{
+			os:      "ming",
```
Comment (alexellis) [inline pkg/get/get_test.go:6741]: We don't tend to have Arm64 Windows apps as a rule.

But let's keep this in for now

File: pkg/get/get_test.go [hunk 2]
```diff
@@ -6703,3 +6703,101 @@ func Test_DownloaSkupper(t *testing.T) {
...
+		toolVersion = "v0.4.0"
+		tool        = getTool(name, tools)
+	)
+
+	tests := []test{
+		{
+			os:      "linux",
+			version: toolVersion,
+			arch:    arch64bit,
+			url:     `https://github.com/kubernetes-sigs/kwok/releases/download/v0.4.0/kwokctl-linux-amd64`,
```
Comment (alexellis) [inline pkg/get/get_test.go:6778]: There is an arm64 Linux binary, but you have to click "Show more" (or similar)

Comment (alexellis) [reply pkg/get/get_test.go:6778]: Screenshot 2023-11-13 at 08 39 39

File: pkg/get/get_test.go (hunk 1 above)
Comment (Shikachuu) [reply pkg/get/get_test.go:6741]: Since they already build them, I think it could be a good addition.

File: pkg/get/get_test.go (hunk 2 above)
Comment (Shikachuu) [reply pkg/get/get_test.go:6778]: Done

**Comment by Shikachuu** [comment]
> @Shikachuu please could you raise an issue with them in their repo?

Seems like they fixed it? I mean this time both local test and the gh action passed.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**